examples/embeddable_examples/public/react_embeddables/search/search_react_embeddable.tsx

```diff
@@ -24,54 +26,39 @@ export const getSearchEmbeddableFactory = (services: Services) => {
     deserializeState: (state) => {
       return state.rawState as State;
     },
-    buildEmbeddable: async (state, buildApi) => {
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const {
+        appliedTimeRange$,
+        cleanupTimeRange,
+        serializeTimeRange,
+        timeRangeApi,
+        timeRangeComparators,
+      } = initializeTimeRange(state, parentApi as Partial<PublishesTimeRange>);
       const defaultDataView = await services.dataViews.getDefaultDataView();
-      const timeRange$ = new BehaviorSubject<TimeRange | undefined>(state.timeRange);
       const dataViews$ = new BehaviorSubject<DataView[] | undefined>(
         defaultDataView ? [defaultDataView] : undefined
       );
       const dataLoading$ = new BehaviorSubject<boolean | undefined>(false);
-      function setTimeRange(nextTimeRange: TimeRange | undefined) {
-        timeRange$.next(nextTimeRange);
-      }
 
       const api = buildApi(
         {
+          ...timeRangeApi,
           dataViews: dataViews$,
-          timeRange$,
-          setTimeRange,
           dataLoading: dataLoading$,
           serializeState: () => {
             return {
               rawState: {
-                timeRange: timeRange$.value,
+                ...serializeTimeRange(),
               },
               references: [],
             };
           },
         },
         {
-          timeRange: [timeRange$, setTimeRange, fastIsEqual],
+          ...timeRangeComparators,
```

Comment:
Nit: If this is the only source of comparators, we don't need the spread.

Reply:
I will leave for consistency with titles initializers unless you feel strongly.